### PR TITLE
Allow collectors to have their individual config files

### PR DESCRIPTION
**Pull request by tsheasha**
Now each collector will have it's own config file that will live in a
config directory. This means each collector will have to be instantiated
and it's config file read into memory. If one collector's config is
corrupt, *ONLY* that collector will be skipped and other non corrupt
ones will start

### CODE REVIEW

File: src/diamond/server.py [hunk 1]
```diff
@@ -105,9 +107,21 @@ def run(self):
...
+                    # will not have that space and needs to have it replaced with an underscore
+                    # instead
+                    config_file = '/'.join([
+                        collectors_config_dir, collector]).replace(' ', '_') + '.conf'
+                    config = load_config(config_file)
+
                     config['enabled'] = True
                     config['fulleritePort'] = self.config['fulleritePort']
                     config['interval'] = config.get('interval', self.config['interval'])
+                    self.collector_config[collector] = config
```
Comment (gnufied) [inline src/diamond/server.py]: Can we add some tests for this please?

**Comment by gnufied** [comment]
also, I think it is probably worth modifying debian packaing to create new directories we expect to exist for storing those collector specific config files.

File: src/diamond/server.py (hunk 1 above)
Comment (tsheasha) [inline src/diamond/server.py]: sure

**Comment by tsheasha** [comment]
modifying debian package is not the way to do it, since the directory for the config is configurable

**Comment by gnufied** [comment]
Ah okay. no worries there then.

**Comment by gnufied** [comment]
:shipit:

File: src/fullerite/collectors.go [hunk 2]
```diff
@@ -6,12 +6,26 @@ import (
...
 	log.Info("Starting collectors...")
-	for name, config := range c.Collectors {
+
+	for _, name := range c.Collectors {
+		configFile := strings.Join([]string{c.CollectorsConfigPath, name}, "/") + ".conf"
+		// Since collector naems can be defined with a space in order to instantiate multiple
+		// instances of the same collector, we want their files
+		// will not have that space and needs to have it replaced with an underscore
+		// instead
+		configFile = strings.Replace(configFile, " ", "_", -1)
```
Comment (tiras-j) [inline src/fullerite/collectors.go]: So we can instantiate multiple collectors with separate configs by specifying in fullerite.conf:
["Test 1", Test 2", "Diamond"] 
and having the appropriate config files under the config directory such as:
/configPath/Test_1.conf
/configPath/Test_2.conf
/configPath/Diamond.conf

That's pretty cool.

**Comment by tiras-j** [comment]
lgtm!

File: src/fullerite/collectors.go (hunk 2 above)
Comment (tsheasha) [inline src/fullerite/collectors.go]: PRETTY cool :)

**Comment by chrisenuf** [comment]
+1

**Comment by poros** [comment]
+1

**Pull request merged**